irods_specific_implementation/irods_rest_logical_path_api_implementation.h

```diff
@@ -0,0 +1,132 @@
+#include <irods/rodsErrorTable.h>
+#include <irods/irods_at_scope_exit.hpp>
+#include <irods/irods_query.hpp>
+
+#include <pistache/http_headers.h>
+#include <pistache/optional.h>
+#include <pistache/router.h>
+
+namespace irods::rest
+{
+    const std::string service_name { "irods_rest_cpp_logicalpath_server" };
+
+    // this is contractually tied directly to the api implementation
+    class logical_path : public api_base
+    {
+    public:
+        logical_path()
+            : api_base{service_name}
+        {
+            info("Endpoint initialized.");
+        }
+
+        std::tuple<Pistache::Http::Code, std::string>
+        rename_dispatcher(const Pistache::Rest::Request& _request,
+                   Pistache::Http::ResponseWriter& _response)
+        {
+            namespace fs = irods::experimental::filesystem;
+
+            try {
+                auto _src = _request.query().get("src").get();
+                auto _dst = _request.query().get("dst").get();
+
+                const fs::path src = decode_url(_src);
+                const fs::path dst = decode_url(_dst);
+
+                auto conn = get_connection(_request.headers().getRaw("authorization").value());
+
+                fs::client::rename(*conn(), src, dst);
+
+                return std::make_tuple(
+                    Pistache::Http::Code::Ok,
+                    ""
+                );
+            }
+            catch (const fs::filesystem_error& e) {
+                error("Caught exception - [error_code={}] {}", e.code().value(), e.what());
+                return make_error_response(e.code().value(), e.what());
+            }
+            catch (const irods::exception& e) {
+                error("Caught exception - [error_code={}] {}", e.code(), e.what());
+                return make_error_response(e.code(), e.what());
+            }
+            catch (const std::exception& e) {
+                error("Caught exception - {}", e.what());
+                return make_error_response(SYS_INTERNAL_ERR, e.what());
+            }
+
+        }
+
+        std::tuple<Pistache::Http::Code, std::string>
+        delete_dispatcher(const Pistache::Rest::Request& _request,
+                   Pistache::Http::ResponseWriter& _response)
+        {
+            namespace fs = irods::experimental::filesystem;
+            namespace fscli = irods::experimental::filesystem::client;
+
+            try {
+                const auto _logical_path = _request.query().get("logical-path").get();
+                const auto _no_trash     = _request.query().get("no-trash").getOrElse("0");
+                const auto _recursive    = _request.query().get("recursive").getOrElse("0");
+                const auto _unregister   = _request.query().get("unregister").getOrElse("0");
+
+                const bool no_trash      = to_bool(_no_trash);
+                const bool recursive     = to_bool(_recursive);
+                const bool unregister    = to_bool(_unregister);
```

Comment:
It looks like nothing else uses these except for the `opts` object on L96. If that is true, then you can move these directly into the initialization of `opts`.

Consider using a different name than `to_bool` or inlining the comparison directly. `to_bool` doesn't help the reader understand what is happening.

Reply:
Recursive is also used in the check on L105, but the other two flags can be made directly.

Reply:
I think this can be resolved?